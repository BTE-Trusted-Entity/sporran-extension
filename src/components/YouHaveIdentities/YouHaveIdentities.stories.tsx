import { Meta } from '@storybook/react';

import {
  identitiesMock,
  IdentitiesProviderMock,
} from '../../utilities/identities/IdentitiesProvider.mock';

import { YouHaveIdentities } from './YouHaveIdentities';

export default {
  title: 'Components/YouHaveIdentities',
  component: YouHaveIdentities,
} as Meta;

export function YouHaveOneIdentity(): JSX.Element {
  return (
    <IdentitiesProviderMock
      identities={{
        '4tDjyLy2gESkLzvaLnpbn7N61VgnwAhqnTHsPPFAwaZjGwP1':
          identitiesMock['4tDjyLy2gESkLzvaLnpbn7N61VgnwAhqnTHsPPFAwaZjGwP1'],
      }}
    >
      <YouHaveIdentities />
    </IdentitiesProviderMock>
  );
}

export function YouHaveMoreIdentities(): JSX.Element {
  return <YouHaveIdentities />;
}
