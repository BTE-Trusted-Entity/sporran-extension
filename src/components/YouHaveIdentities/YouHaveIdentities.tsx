import { Fragment } from 'react';

import { useIdentities } from '../../utilities/identities/identities';
import { plural } from '../../utilities/plural/plural';

export function YouHaveIdentities(): JSX.Element | null {
  const identities = useIdentities().data;
  if (!identities) {
    // empty string to force rendering of the containing component and prevent layout shift
    return <Fragment> </Fragment>; // storage data pending
  }

  const identitiesNumber = Object.values(identities).length;

  return (
    <Fragment>
      {plural(identitiesNumber, {
        one: 'component_YouHaveIdentities_one',
        other: 'component_YouHaveIdentities_other',
      })}
    </Fragment>
  );
}
