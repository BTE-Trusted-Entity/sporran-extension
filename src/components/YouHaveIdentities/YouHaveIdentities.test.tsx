import {
  identitiesMock,
  IdentitiesProviderMock,
  render,
} from '../../testing/testing';

import { YouHaveIdentities } from './YouHaveIdentities';

describe('YouHaveIdentities', () => {
  it('should match the snapshot for one identity', async () => {
    const { container } = render(
      <IdentitiesProviderMock
        identities={{
          '4tDjyLy2gESkLzvaLnpbn7N61VgnwAhqnTHsPPFAwaZjGwP1':
            identitiesMock['4tDjyLy2gESkLzvaLnpbn7N61VgnwAhqnTHsPPFAwaZjGwP1'],
        }}
      >
        <YouHaveIdentities />
      </IdentitiesProviderMock>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot for multiple identities', async () => {
    const { container } = render(<YouHaveIdentities />);
    expect(container).toMatchSnapshot();
  });
});
