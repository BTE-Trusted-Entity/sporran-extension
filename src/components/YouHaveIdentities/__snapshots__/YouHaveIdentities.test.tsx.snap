// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YouHaveIdentities should match the snapshot for multiple identities 1`] = `
<div>
  You have 3 Identities
</div>
`;

exports[`YouHaveIdentities should match the snapshot for one identity 1`] = `
<div>
  You have 1 Identity
</div>
`;
